### PR TITLE
Expand image_processing_demo to work with EDSR like SR models

### PR DESCRIPTION
**Pull request by TolyaTalamanov**

### CODE REVIEW

File: demos/common/cpp/models/src/model_base.cpp [hunk 1]
```diff
@@ -34,9 +34,6 @@ std::shared_ptr<ov::Model> ModelBase::prepareModel(ov::Core& core) {
     // -------------------------- Reading all outputs names and customizing I/O tensors (in inherited classes)
     prepareInputsOutputs(model);
 
-    /** Set batch size to 1 **/
-    ov::set_batch(model, 1);
```
Comment (TolyaTalamanov) [inline demos/common/cpp/models/src/model_base.cpp:38]: @Wovchena Since `MO` can return IR with batch, should we move to separate `setBatch` method?

**Comment by TolyaTalamanov** [comment]
@Wovchena Could you have a look, please?

File: demos/image_processing_demo/cpp/main.cpp [hunk 2]
```diff
@@ -57,7 +57,8 @@ DEFINE_INPUT_FLAGS
 DEFINE_OUTPUT_FLAGS
 
 static const char help_message[] = "Print a usage message.";
-static const char at_message[] = "Required. Type of the model, either 'sr' for Super Resolution task, 'deblur' for "
+static const char at_message[] = "Required. Type of the model, either 'sr' for Super Resolution task, "
+                                 "'sr_channel_joint' for Super Resolution task that accepts and returns 1 channel image, 'deblur' for "
```
Comment (TolyaTalamanov) [inline demos/image_processing_demo/cpp/main.cpp:61]: `task` -> `model` ???

Comment (Wovchena) [reply demos/image_processing_demo/cpp/main.cpp:61]: Yes. Please change it. Also please update README Running section after that

File: demos/common/cpp/models/src/model_base.cpp (hunk 1 above)
Comment (Wovchena) [reply demos/common/cpp/models/src/model_base.cpp:38]: Alternative solution could be to modify `prepareInputsOutputs()` for every model wrapper. I don't see any reason to prefer one or another. Let's proceed with your suggestion to introduce `setBatch` method.

File: demos/image_processing_demo/cpp/main.cpp (hunk 2 above)
Comment (TolyaTalamanov) [reply demos/image_processing_demo/cpp/main.cpp]: Done

File: demos/common/cpp/models/src/model_base.cpp (hunk 1 above)
Comment (TolyaTalamanov) [reply demos/common/cpp/models/src/model_base.cpp:38]: Done

**Review by Wovchena — APPROVED** [review]
Is it urgent to merge? I'm not sure when the problem with model will be resolved

**Pull request merged**